packages/aws-cdk-lib/aws-cloudwatch/lib/metric.ts

```diff
@@ -723,6 +748,31 @@ export class MathExpression implements IMetric {
     });
   }
 
+  /**
+   * Make a new Alarm for this metric using anomaly detection
+   *
+   * @param scope The scope in which to create the alarm
+   * @param id The ID of the alarm
+   * @param props Alarm creation properties for anomaly detection
+   */
+  public createAnomalyDetectionAlarm(scope: Construct, id: string, props: CreateAnomalyDetectionAlarmProps): Alarm {
+    if (!Alarm.isAnomalyDetectionOperator(props.comparisonOperator)) {
+      throw new Error(`Invalid comparison operator for anomaly detection alarm: ${props.comparisonOperator}`);
+    }
+    const anomalyDetectionExpression = new MathExpression({
+      expression: `ANOMALY_DETECTION_BAND(m0, ${props.bounds ?? 1})`,
+      usingMetrics: { m0: this },
+      period: props.period,
+    });
+
+    return new Alarm(scope, id, {
+      ...props,
+      metric: anomalyDetectionExpression,
+      threshold: 0, // This value will be ignored for anomaly detection alarms
+      thresholdMetricId: 'expr_1',
+    });
+  }
+
```

Comment:
Looks like this function is identical to the one in Metric class. Is there a way to re-use the same function?

Reply:
Correct I will factorize that part :)